packages/babel-traverse/src/path/index.js

```diff
@@ -138,44 +156,143 @@ export default class NodePath {
     let path = this;
     do {
       let key = path.key;
+      // $FlowFixMe yeah yeah: in theory these can be null
       if (path.inList) key = `${path.listKey}[${key}]`;
       parts.unshift(key);
     } while ((path = path.parentPath));
     return parts.join(".");
   }
 
-  debug(message) {
+  debug(message: any) {
     if (!debug.enabled) return;
+    // $FlowFixMe yeah yeah: in theory these can be null
     debug(`${this.getPathLocation()} ${this.type}: ${message}`);
   }
 
   toString() {
     return generator(this.node).code;
   }
-}
 
-Object.assign(
-  NodePath.prototype,
-  NodePath_ancestry,
-  NodePath_inference,
-  NodePath_replacement,
-  NodePath_evaluation,
-  NodePath_conversion,
-  NodePath_introspection,
-  NodePath_context,
-  NodePath_removal,
-  NodePath_modification,
-  NodePath_family,
-  NodePath_comments,
```

Comment:
I feel like there's a sensible way to refactor this mixin pattern. It might be worth asking someone more adept with flow for suggestions. For now, I used a bit of grep and sed to generate the below and pasted it in.

The `is${type}` and `assert${type}` functions are going to want some kind of generated file.

I wonder if we could do make a module that does `export * from $file` for each of the above modules (plus some generated files with the is* definitions in, and then somehow `extend` from that?